spec/features/service_status_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'capybara_helper'
+
+describe 'viewing service status' do
+  let(:user) { User.find_or_create_by(name: 'test user', email: 'test@example.justice.gov.uk') }
+  let(:service) do
+    Service.create!(name: 'ABC Service', git_repo_url: 'https://github.com/some-org/some-repo.git',
+                    created_by_user: user)
+  end
+  before do
+    login_as!(user)
+  end
+
+  describe 'no service status checks' do
+    context 'with no completed deployments' do
+      before do
+        visit "/services/#{service.slug}"
+      end
+      it 'does not show a status' do
+        expect(page).not_to have_selector('span.status')
+        expect(page).to have_selector('span', text: I18n.t('services.environment.no_deployment'))
+      end
+      it 'does not show a `Check now` button' do
+        expect(page).not_to have_button(I18n.t('services.environment.check_now'))
+      end
+    end
+    context 'with completed deployments' do
+      before do
+        completed_deployment
+        visit "/services/#{service.slug}"
+      end
+      it 'does not show a status' do
+        expect(page).not_to have_selector('span.status')
+        expect(page).to have_selector('span', text: I18n.t('services.environment.no_deployment'))
+      end
+
+      it 'does show a `Check now` button' do
+        expect(page).to have_button(I18n.t('services.environment.check_now'))
+      end
+    end
+  end
+
+  describe 'service status checks' do
+    context 'with no deployments' do
+      before do
+        ServiceStatusCheck.create!(environment_slug: 'dev', status: 404, time_taken: 30.0,
```

Comment:
logically, this `ServiceStatusCheck.create!` should probably be in a separate `before` block directly under the `describe 'service status checks' do`